db/Pos/DB/Functions.hs

```diff
@@ -41,6 +43,23 @@ dbGetBi tag key = do
 dbPutBi :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> m ()
 dbPutBi tag k v = dbPut tag k (serialize' v)
 
+-- | Read serialized value (with version) associated with given key from pure DB.
+dbGetBiWithVersion
+    :: forall v m.
+       (Bi v, MonadDBRead m)
+    => DBTag -> ByteString -> m (Maybe (Word8, v))
+dbGetBiWithVersion tag key = do
+    bytes <- dbGet tag key
+    traverse (dbDecode . (ToDecodeValue key)) bytes
+
+-- | Write serializable value to DB for given key. Uses simple versioning.
+dbPutBiWithVersion :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> Word8 -> m ()
+dbPutBiWithVersion tag k v version = dbPut tag k (dbSerialize version v)
```

Comment:
Shouldn't we check somehow that the previous version of the data (if it exists) has version `version-1`?